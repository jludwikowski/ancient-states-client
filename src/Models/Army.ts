import {Entity} from './Entity';
import Leader from './Leader';
import UnitReference from "./UnitReference";

export default interface Army extends Entity {
    commander: Leader | null,
    units: UnitReference[]
    owner: number,
    position: string,
    status: string,
    orderDate?: string | null,
    targetId?: number | null,
}
