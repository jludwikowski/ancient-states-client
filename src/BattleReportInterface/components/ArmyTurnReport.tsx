import React from "react";
import Army from "../../Models/Army";

interface Props {
    army: Army
}

export const ArmyTurnReport: React.FunctionComponent<Props> = props => (
    <li className="list-group-item">
        <span>{props.army.name}</span>
        <span className="unit">
        {props.army.units.map((unit, index) => (
            <React.Fragment key={index}>
                <div>UnitId: {unit.baseUnit}</div>
                <div>Level: {unit.level}</div>
                <div>Number: {unit.number}</div>
            </React.Fragment>
        ))}
        </span>
    </li>
)
