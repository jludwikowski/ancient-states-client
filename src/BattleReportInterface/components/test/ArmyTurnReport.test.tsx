import { shallow } from 'enzyme';
import React from 'react';
import ReactDOM from 'react-dom';
import Army from '../../../Models/Army';
import { ArmyTurnReport } from '../ArmyTurnReport';

describe('ArmyTurnReport', () => {

    const army:Army = {
        id: 7,
        name: 'New Army',
        description: '',
        status: 'camp',
        orderDate: null,
        targetId: null,
        position: '0,0',
        owner: 1,
        commander: null,
        units: [
            {
                number: 9802,
                level: 1,
                baseUnit: 1,
            }
        ]
    };

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<ArmyTurnReport army={army} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('Should match snapshot', () => {
        const component = shallow(<ArmyTurnReport army={army} />);
        expect(component).toMatchSnapshot();
    });
});
