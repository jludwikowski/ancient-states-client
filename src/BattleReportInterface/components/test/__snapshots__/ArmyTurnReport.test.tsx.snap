// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArmyTurnReport Should match snapshot 1`] = `
<li
  className="list-group-item"
>
  <span>
    New Army
  </span>
  <span
    className="unit"
  >
    <div>
      UnitId: 
      1
    </div>
    <div>
      Level: 
      1
    </div>
    <div>
      Number: 
      9802
    </div>
  </span>
</li>
`;
